src/v/rpc/parse_utils.h

```diff
@@ -275,25 +276,25 @@ ss::future<T> parse_type(ss::input_stream<char>& in, const header& h) {
     return read_iobuf_exactly(in, h.payload_size).then([h](iobuf io) {
         validate_payload_and_header(io, h);
 
-        switch (h.compression) {
-        case compression_type::none:
-            break;
-
-        case compression_type::zstd: {
-            compression::stream_zstd fn;
-            io = fn.uncompress(std::move(io));
-            break;
-        }
-
-        default:
-            return ss::make_exception_future<T>(std::runtime_error(
+        auto iobuf_fut = ss::do_with(std::move(io), [h](iobuf& io) {
```

Comment:
ISTM we can just avoid this `do_with` entirely if we can move `io` into the compression method.

Reply:
Yep! removing the `do_with`